examples/dapperish.go

```diff
@@ -19,10 +18,10 @@ import (
 func client() {
 	reader := bufio.NewReader(os.Stdin)
 	for {
-		span := opentracing.StartTrace("getInput")
-		ctx := opentracing.BackgroundGoContextWithSpan(span)
+		ctx, span := opentracing.BackgroundContextWithSpan(
+			opentracing.StartTrace("getInput"))
 		// Make sure that global trace tag propagation works.
-		span.TraceContext().SetTraceAttribute("User", os.Getenv("USER"))
+		span.SetTraceAttribute("User", os.Getenv("USER"))
```

Comment:
:+1: -- `Attribute` vs `Tag` was a strange terminology distinction when I first started getting into this project; nothing unusual but the kind of naming that software is unfortunately rife with.  So I think it can be clearer.

However, I'm not sure folding `Attributes` into `Span Tags` (via `CascadingTag` or similar) is going to solve the problem.  A `Tag` is data added to a span, whereas an `Attribute` is data added to a `Context`.  Based on that `Attributes` feel like they are not part of `Span` creation.  We do want to propagate them, but they are not `Span`-centric.

It makes sense to think of the relevant part of `DCP` as propagating a `Span` when you have a zipkin-like model where client and server report data into the same `Span`, but when considering more generalized tracing systems, `Context` is the thing that is propagated, and `Span`s are created to report information on execution as the `Context` passes through.  We can add `Span` info to the `Context` before propagating to inform the structure of the tree and `Span`s, but that doesn't make the `Context` a `Span`.

I realize this is arguably scope question currently being debated in #33 as well, but clarity is the goal of this line of discussion and I think it will be clearer if we don't conflate these two concepts.